test/extended/cmd.sh

```diff
@@ -74,75 +75,74 @@ popd
+os::cmd::expect_success_and_text 'oc describe bc' 'From Image:\s+DockerImage test/scratchimage:latest'
+os::cmd::expect_success 'oc delete project test-scratchimage'
+os::cmd::expect_success "oc project ${current_project}"
 # error due to partial match
-[ "$(oc new-app test/scratchimage2 -o yaml |& tr '\n' ' ' 2>&1 | grep -E "partial match")" ]
+os::cmd::expect_failure_and_text "oc new-app test/scratchimage2 -o yaml" "partial match"
 # success with exact match	
-[ "$(oc new-app test/scratchimage -o yaml)" ]
+os::cmd::expect_success "oc new-app test/scratchimage"
 echo "[INFO] newapp: ok"
 
 echo "[INFO] Running env variable expansion tests"
-oc new-project envtest
-oc create -f test/extended/fixtures/test-env-pod.json
-tryuntil "oc get pods | grep Running"
+os::cmd::expect_success 'oc new-project envtest'
+os::cmd::expect_success 'oc create -f test/extended/fixtures/test-env-pod.json'
+os::cmd::try_until_text 'oc get pods' 'Running'
 podname=$(oc get pods --template='{{with index .items 0}}{{.metadata.name}}{{end}}')
-oc exec test-pod env | grep podname=test-pod
-oc exec test-pod env | grep podname_composed=test-pod_composed
-oc exec test-pod env | grep var1=value1
-oc exec test-pod env | grep var2=value1
-oc exec test-pod ps ax | grep "sleep 120"
+os::cmd::expect_success_and_text 'oc exec test-pod env' 'podname=test-pod'
+os::cmd::expect_success_and_text 'oc exec test-pod env' 'podname_composed=test-pod_composed'
+os::cmd::expect_success_and_text 'oc exec test-pod env' 'var1=value1'
+os::cmd::expect_success_and_text 'oc exec test-pod env' 'var2=value1'
+os::cmd::expect_success_and_text 'oc exec test-pod ps ax' 'sleep 120'
 echo "[INFO] variable-expansion: ok"
 
 echo "[INFO] Running image pull secrets tests"
-oc login ${MASTER_ADDR} -u pull-secrets-user -p password --certificate-authority=${MASTER_CONFIG_DIR}/ca.crt
+VERBOSE=true os::cmd::expect_success "oc login ${MASTER_ADDR} -u pull-secrets-user -p password --certificate-authority=${MASTER_CONFIG_DIR}/ca.crt"
 
 # create a new project and push a busybox image in there
-oc new-project image-ns
-oc delete all --all
+os::cmd::expect_success 'oc new-project image-ns'
+os::cmd::expect_success 'oc delete all --all'
 IMAGE_NS_TOKEN=`oc get sa/builder --template='{{range .secrets}}{{ .name }} {{end}}' | xargs -n 1 oc get secret --template='{{ if .data.token }}{{ .data.token }}{{end}}' | base64 -d -`
-docker login -u imagensbuilder -p ${IMAGE_NS_TOKEN} -e fake@example.org ${DOCKER_REGISTRY}
-oc tag --source=docker busybox:latest image-ns/busybox:latest
-oc import-image busybox
-docker pull busybox
-docker tag -f docker.io/busybox:latest ${DOCKER_REGISTRY}/image-ns/busybox:latest
-docker push ${DOCKER_REGISTRY}/image-ns/busybox:latest
-docker rmi -f ${DOCKER_REGISTRY}/image-ns/busybox:latest
+os::cmd::expect_success "docker login -u imagensbuilder -p ${IMAGE_NS_TOKEN} -e fake@example.org ${DOCKER_REGISTRY}"
```

Comment:
Intermittently getting

```
FAILURE after 4.030s: test/extended/cmd.sh:109: executing 'docker login -u imagensbuilder -p <snip> -e fake@example.org 172.30.52.246:5000' expecting success: the command returned the wrong error code
There was no output from the command.
Standard error from the command:
Error response from daemon: invalid registry endpoint "http://172.30.52.246:5000/v0/". HTTPS attempt: unable to ping registry endpoint https://172.30.52.246:5000/v0/
v2 ping attempt failed with error: Get https://172.30.52.246:5000/v2/: dial tcp 172.30.52.246:5000: connection refused
 v1 ping attempt failed with error: Get https://172.30.52.246:5000/v1/_ping: dial tcp 172.30.52.246:5000: connection refused. HTTP attempt: unable to ping registry endpoint http://172.30.52.246:5000/v0/
v2 ping attempt failed with error: Get http://172.30.52.246:5000/v2/: dial tcp 172.30.52.246:5000: connection refused
 v1 ping attempt failed with error: Get http://172.30.52.246:5000/v1/_ping: dial tcp 172.30.52.246:5000: connection refused
```

Master is flaking on startup so I'm getting stuck figuring out why - @bparees thoughts?